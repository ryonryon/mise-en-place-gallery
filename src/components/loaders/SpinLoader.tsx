import styled from "@emotion/styled";
import * as React from "react";
import { secondary, white } from "../../constants/colors";

interface Props {
  size: string;
  color: string;
  coverColor: string;
}

const SpinLoader = ({ size, color, coverColor }: Props) => {
  const Spinner = styled.div`
    border: 16px solid ${coverColor};
    border-top: 16px solid ${color};
    border-radius: 50%;
    width: ${size};
    height: ${size};
    animation: spin 2s linear infinite;

    @keyframes spin {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
  `;
  return <Spinner />;
};

SpinLoader.defaultProps = {
  size: "70px",
  color: secondary,
  coverColor: white
};

export default SpinLoader;
