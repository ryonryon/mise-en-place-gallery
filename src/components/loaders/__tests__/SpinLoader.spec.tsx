import * as React from "react";
import * as renderer from "react-test-renderer";

import SpinLoader from "../SpinLoader";

describe("SpinLoader test", () => {
  it("tests SpinLoader test render correctly", () => {
    const spinLoader = renderer.create(<SpinLoader />).toJSON();

    expect(spinLoader).toMatchSnapshot();
  });
});
