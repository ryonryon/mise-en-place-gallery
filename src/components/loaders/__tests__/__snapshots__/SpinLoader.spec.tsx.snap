// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinLoader test tests SpinLoader test render correctly 1`] = `
<div
  className="css-q816ym"
/>
`;
